pallets/settlement/src/lib.rs

```diff
@@ -775,6 +778,22 @@ decl_module! {
             Self::deposit_event(RawEvent::VenuesBlocked(did, ticker, venues));
         }
 
+        /// Marks a receipt issued by the caller as claimed or not claimed.
+        /// This allows the receipt issuer to invalidate an already issued receipt or revalidate an already claimed receipt.
+        ///
+        /// * `receipt_uid` - Unique ID of the receipt.
+        /// * `validity` - New validity of the receipt.
+        #[weight = <T as Trait>::WeightInfo::change_receipt_validity()]
+        pub fn change_receipt_validity(origin, receipt_uid: u64, validity: bool) {
+            let PermissionedCallOriginData {
+                primary_did,
+                sender: signer,
+                ..
+            } = Identity::<T>::ensure_origin_call_permissions(origin)?;
+            <ReceiptsUsed<T>>::insert(&signer, receipt_uid, !validity);
```

Comment:
Why the validity is stored in a negated form while it is emitted in the original form in the event?

I think it should be stored in the original form as well.

Reply:
The storage is for `ReceiptsUsed`, not for Validity. By default, all receipts are unused and we treat unused receipts as Valid.

I didn't want to do a storage migration to change the meaning.